### PR TITLE
Add `make-rectangular`, `make-polar` for functions, numbers, symbolic types

### PR DESCRIPTION
**Pull request by sritchie**
This PR:

- adds a `sicmutils.complex/I` alias for `i`
- Adds the following two generic operators:
  - `g/make-rectangular`, (build a complex number from real and imaginary parts)
  - `g/make-polar` (build a complex number from radius and angle)

these work with real numbers, symbolic numbers, functions and any combination of these. They match the scmutils behavior in all cases.

It WOULD be nice to further extend these to handle up, down and matrices, for constructing complex structures/matrices where the shapes match.

### CODE REVIEW

File: src/sicmutils/numbers.cljc [hunk 1]
```diff
@@ -58,7 +58,13 @@
 ;; ## Complex Operations
 (defmethod g/real-part [::v/real] [a] a)
 (defmethod g/imag-part [::v/real] [a] 0)
-(defmethod g/angle [::v/real] [a] (v/zero-like a))
+
+(let [neg-pi (core-minus Math/PI)]
```
Comment (sritchie) [inline src/sicmutils/numbers.cljc:62]: I know I had this in a previous PR and didn't push it forward. I'll deal with the merge when I get there...

File: test/sicmutils/abstract/number_test.cljc [hunk 2]
```diff
@@ -499,6 +515,52 @@
...
+            (is (v/= (g/+ sym (g/* c/I n))
+                     (g/make-rectangular sym n))))
+
+  (checking "make-polar" 100 [r     gen/symbol
+                              theta gen/symbol]
+            (is (= (g/* r (g/+ (g/cos theta)
+                               (g/* c/I (g/sin theta))))
+                   (g/make-polar r theta))))
+
+  (checking "make-polar with numbers" 100
```
Comment (littleredcomputer) [inline test/sicmutils/abstract/number_test.cljc:537]: it is so nice to have tests like these

**Review by littleredcomputer — COMMENTED** [review]
awesome

File: test/sicmutils/abstract/number_test.cljc (hunk 2 above)
Comment (sritchie) [reply test/sicmutils/abstract/number_test.cljc:537]: This project has really gotten me used to leaning ONLY on tests like this and being less comfortable with only bare unit tests. A win!

**Pull request merged**